src/borrow_tracker/tree_borrows/perms.rs

```diff
@@ -271,14 +265,93 @@ pub mod diagnostics {
         pub fn summary(&self) -> &'static str {
             assert!(Self::is_possible(self.0, self.1));
             match (self.0, self.1) {
-                (_, Active) => "an activation",
+                (_, Active) => "the first write to a 2-phase borrowed mutable reference",
                 (_, Frozen) => "a loss of write permissions",
                 (Frozen, Disabled) => "a loss of read permissions",
                 (_, Disabled) => "a loss of read and write permissions",
                 (old, new) =>
                     unreachable!("Transition from {old:?} to {new:?} should never be possible"),
             }
         }
+
+        /// Not all transitions are relevant to all errors, this function filters those out.
+        ///
+        /// For example we don't care about the activation of a pointer when the error is
+        /// because it is not readable. All we care about is when it lost read permissions.
+        /// Return value to be interpreted as
+        /// - `None`: not a relevant transition
+        /// - `Some(v)`: relevant transition, with the additional help messages `v`
+        ///
+        /// Are considered unreachable
+        /// - any errors that involve noop transitions, as those can never be the cause of the error
+        /// - other configurations that are known to never cause errors (e.g. `Active` cannot cause a
+        /// `ChildAccessForbidden` error because all child accesses are allowed on `Active`)
+        /// - violations of `<` for permissions: `self` is assumed to be anterior to `err`
```

Comment:
What does 'anterior' mean here?

Reply:
The transition represented by `self` should occur before the error, so we're never considering things like `self = (Frozen, Disabled)` while `err = ProtectedTransition(Active, Frozen)` because that would imply that somewhere a permission did `Frozen -> Disabled -> ??? -> Active -> Frozen` which is impossible.